### ISSUE TITLE
focused terminal turns invisible with chunkwm when opening another application

### ISSUE THREAD

**Issue by jwkvam** [jwilm/alacritty#1125]
Which operating system does the issue occur on? MacOS 10.13.3
Alacritty version: 03f9e0cd9a417550d53f34f7b9671a488363d411

A focused alacritty terminal turns invisible when another app is opened. I'm using [chunkwm](https://github.com/koekeishiya/chunkwm) (a tiling window manager) so the alacritty terminal gets resized when I open another app. The contents of the alacritty terminal become visible when I focus the window again. I attached a gif showing this in action, the red outline shows which window has focus.

If I had to guess, this weird combination of a window being resized but losing focus that causes the terminal to stay invisible.

This is really just an annoyance :)

![alac](https://user-images.githubusercontent.com/86304/36497669-ce8a28ce-16f0-11e8-970a-3477855cbdb9.gif)

**Comment by chrisduerr** [comment]
Hmm If I had to guess the window probably isn't redrawn even though it should be.

However there should be a redraw issued after resizing, not sure why it doesn't happen. There have been some issues with transparency in the past, would you mind testing this with transparency disabled?

You can disable it by setting the `true` ![here](https://github.com/jwilm/alacritty/blob/master/src/window.rs#L200) to `false`.

**Comment by jwkvam** [comment]
I still see the same behavior with that change. FWIW I have transparency off in my config `background_opacity: 1.0`. I also tried running alacritty with `-vvv` and this is what I see:

There's no output when the terminal gets resized and turns invisible. I see the following output when I focus the terminal.

```
Term::resize
num_cols, num_lines = 105, 57
width: 640, height: 698
Set Attribute: Foreground(Named(Black))
designate G0 character set as Ascii
Set Attribute: Reset
carriage_return
carriage_return
Set Attribute: Bold
Set Attribute: Foreground(Named(White))
Set Attribute: Background(Named(Green))
Set Attribute: Foreground(Named(Black))
designate G0 character set as Ascii
Set Attribute: Reset
Set Attribute: Foreground(Named(Black))
designate G0 character set as Ascii
Set Attribute: Reset
Set Attribute: Foreground(Named(Black))
designate G0 character set as Ascii
Set Attribute: Reset
Set Attribute: Foreground(Named(Green))
Set Attribute: Foreground(Named(Black))
designate G0 character set as Ascii
Set Attribute: Reset
Set Attribute: Bold
Set Attribute: Foreground(Named(Magenta))
Set Attribute: Foreground(Named(Black))
designate G0 character set as Ascii
Set Attribute: Reset
Set Attribute: Foreground(Named(Blue))
Set Attribute: Foreground(Named(Black))
designate G0 character set as Ascii
Set Attribute: Reset
Set Attribute: Foreground(Named(Black))
designate G0 character set as Ascii
Set Attribute: Reset
Set Attribute: Foreground(Named(Black))
designate G0 character set as Ascii
Set Attribute: Reset
Set Attribute: Foreground(Named(Black))
designate G0 character set as Ascii
Set Attribute: Reset
clear_screen: Below
move_forward: 58
carriage_return
move_forward: 95
carriage_return
move_forward: 37
```

**Comment by chrisduerr** [comment]
Looks like it's not receiving resize events until the window is focused. That's a bit odd.

**Comment by jwilm** [comment]
@chrisduerr it's due to the macOS event loop model and winit's implementation. There's a PR in the pipeline which runs the event loop on a coroutine which is then suspended when receiving resize events thereby allowing us to redraw as it's received.

https://github.com/tomaka/winit/pull/237

**Comment by jwkvam** [comment]
Hmm seems like the author of that PR got frustrated :(

**Comment by daGrevis** [comment]
I have exactly the same problem. Waiting for [winit PR](https://github.com/tomaka/winit/pull/237).

**Comment by tubbo** [comment]
This happens to me as well, but I'm running **Amethyst v0.12.2**

**Comment by Ciebiada** [comment]
Issue is still there with the newest stable Alacritty and Amethyst (0.3 and 0.13). Any updates?

**Comment by chrisduerr** [comment]
If you're interested in tracking progress, I'd recommend subscribing to the upstream issue.

I'm struggling with this myself on macOS, however there's nothing Alacritty can do here. Once it's fixed upstream, it will be resolved in Alacritty within a short period of time.

**Comment by Ciebiada** [comment]
Thanks

**Comment by alexmaras** [comment]
It looks like that PR has been closed due to the creation of EventLoop 2.0.

**Comment by nixpulvis** [comment]
@alexmaras you can follow the work being done on the new event loop integration in #2438. I'd be curious to hear if this solves the issue @jwkvam described.

**Comment by kchibisov** [comment]
@nixpulvis `run_return` is steel not implemented in `winit` on macOs, so you can't run #2438 on macOs atm. For reference #1005

**Comment by chrisduerr** [comment]
I've noticed that https://github.com/jwilm/alacritty/pull/2438 seems to fix this issue. If anyone is interested in confirming this for other WMs (I've been using yabai), please let me know.
